hera_mc/cm_sysutils.py

```diff
@@ -424,3 +424,299 @@ def get_apriori_antenna_status_for_rtp(self, status, at_date='now'):
+        node = geo_sysdef.read_nodes()
+        for this_node in node.keys():
+            node_hpn = 'N{:02d}'.format(this_node)
+            ants_per_node[node_hpn] = []
+            for ant in node[this_node]['ants']:
+                if ant in geo_sysdef.region['heraringa']:
+                    prefix = 'HA'
+                elif ant in geo_sysdef.region['heraringb']:
+                    prefix = 'HB'
+                else:
+                    prefix = 'HH'
+                ants_per_node[node_hpn].append("{}{}".format(prefix, ant))
+    else:
+        if isinstance(source, str) and source.lower().startswith('h'):
+            source = cm_hookup.Hookup(session=session)
+        hu_dict = source.get_hookup(cm_sysdef.hera_zone_prefixes, hookup_type='parts_hera')
+        for this_ant, vna in hu_dict.items():
+            key = vna.hookup['E<ground'][-1].downstream_part
+            if key[0] != 'N':
+                continue
+            ants_per_node.setdefault(key, [])
+            ants_per_node[key].append(cm_utils.split_part_key(this_ant)[0])
+    return ants_per_node
+
+
+def _get_dict_elements(npk, hu, ele_a, ele_b):
+    """Return the appropriate hookup elements for node_info."""
+    a = ele_a.lower()
+    b = ele_b.lower()
+    A = ele_a.upper()
+    B = ele_b.upper()
+    e_ret = {a: '', b: ''}
+    try:
+        e_hookup = hu[npk].hookup['@<middle']
+    except KeyError:
+        return e_ret
+    for element in e_hookup:
+        if element.upstream_part.startswith(A):
+            e_ret[a] = element.upstream_part
+            e_ret[b] = element.downstream_part
+            break
+        elif element.upstream_part.startswith(B):
+            e_ret[b] = element.upstream_part
+    return e_ret
+
+
+def _find_ant_node(pnsearch, na_dict):
+    found_node = None
+    for node, antennas in na_dict.items():
+        for ant in antennas:
+            antint = cm_utils.peel_key(ant, 'NPR')[0]
+            if pnsearch == antint:
+                if found_node is not None:
+                    raise ValueError("Antenna {} already listed in node {}"
+                                     .format(pnsearch, found_node))
+                else:
+                    found_node = node
+    return found_node
+
+
+def which_node(ant_num, session=None):
+    """
+    Find node for antenna.
+
+    Parameters
+    ----------
+    ant_num : int or list of int or csv-list or hyphen-range str
+        Antenna numbers, as int
+
+    Returns
+    -------
+    dict
+        Contains antenna and node
+    """
+    na_from_file = node_antennas('file', session=session)
+    na_from_hookup = node_antennas('hookup', session=session)
+    ant_num = cm_utils.listify(ant_num)
+    ant_node = {}
+    for pn in ant_num:
+        pnint = cm_utils.peel_key(str(pn), 'NPR')[0]
+        ant_node[pnint] = [_find_ant_node(pnint, na_from_file)]
+        ant_node[pnint].append(_find_ant_node(pnint, na_from_hookup))
+    return ant_node
+
+
+def print_which_node(ant_node):
```

Comment:
I find this function name confusing -- it sounds like it prints something but it doesn't, it returns a formatted string.

Reply:
I struggle with this - does the module "print" or return a print string -- the latter is preferable to feed text to e.g. Matt K.  Any advice on what to rename for the distinction?  Ideally _everything_ would do it consistently, but I don't want to go over the entire codebase.

Reply:
how about "format_which_node" or "string_which_node" for things that return strings, reserving "print" for methods that actually print things.

Reply:
How about this belt and suspenders approach?
def print_which_node(ant_node):
    print(formatted__which_node__string(ant_node))


def formatted__which_node__string(ant_node):

Reply:
it's a little silly but it is at least clear.